src/main/scala/com/microsoft/hyperspace/index/sources/delta/DeltaLakeRelation.scala

```diff
@@ -36,9 +37,9 @@ class DeltaLakeRelation(spark: SparkSession, override val plan: LogicalRelation)
   /**
    * Computes the signature of the current relation.
    */
-  override def signature: String = plan.relation match {
+  override def signature(fb: FingerprintBuilder): Option[Fingerprint] = plan.relation match {
     case HadoopFsRelation(location: TahoeLogFileIndex, _, _, _, _, _) =>
-      location.tableVersion + location.path.toString
+      Some(fb.add(location.tableVersion).add(location.path.toString).build())
```

Comment:
If it's always retuning only `Some` let's remove it and return `Fingerprint` directly.

Reply:
Since it is an overriding method, you can't change the return type.